packages/codecs-core/src/reverse-codec.ts

```diff
@@ -17,8 +24,11 @@ export function reverseEncoder<TFrom, TSize extends number>(
     try {
         assertIsFixedSize(encoder);
     } catch (e) {
-        // TODO: Coded error, also proper catch handling
-        throw new Error('Cannot reverse a codec of variable size.');
+        if (isSolanaError(e) && e.context.__code === SOLANA_ERROR__CODECS_EXPECTED_FIXED_LENGTH_GOT_VARIABLE_LENGTH) {
+            throw new SolanaError(SOLANA_ERROR__CODECS_CANNOT_REVERSE_CODEC_OF_VARIABLE_SIZE);
+        } else {
+            throw e;
+        }
```

Comment:
Checking `__code_` is essentially what `isSolanaError` does.

```suggestion
        if (isSolanaError(e, SOLANA_ERROR__CODECS_EXPECTED_FIXED_LENGTH_GOT_VARIABLE_LENGTH)) {
            throw new SolanaError(SOLANA_ERROR__CODECS_CANNOT_REVERSE_CODEC_OF_VARIABLE_SIZE);
        } else {
            throw e;
        }
```

Reply:
Bah, I did this elsewhere but must have missed these.